message.go

```diff
@@ -100,6 +101,9 @@ func (m *Message) Release() {
 func (m *Message) Reset(arena Arena) (first *Segment, err error) {
 	m.capTable.Reset()
 	for k := range m.segs {
+		if k == 0 && m.segs[k] == &m.firstSeg {
```

Comment:
A comment might be helpful here.  Something along the lines of:

```go
// optimization:  keep the first segment so that the re-used Message does not have to
// allocate a new one.
```